### DISCUSSION TITLE
MAE train & MAE test loss functions give same values over multiple epochs, effectively no training being done.

### DISCUSSION THREAD

**Discussion by ethan530** [mrdbourke/pytorch-deep-learning#311] [Q&A] [answered]
Most of the code is from the pytorch_workflow.ipynb file, I changed the dataset to be google stock data I found online. 
[Google.csv](https://github.com/mrdbourke/pytorch-deep-learning/files/10800758/Google.csv)

In the zip file is the code I have written.
[v2.zip](https://github.com/mrdbourke/pytorch-deep-learning/files/10800812/v2.zip)

If you run through the cells and get to the loss calculation, the test and training loss is:
<img width="422" alt="image" src="https://user-images.githubusercontent.com/74956050/220544558-debc4eb3-8eab-4fcb-94b2-f6e4e01bcce1.png">

Even if I change the number of epochs to a higher number, the calculations do not change, they remain at the same number instead of decreasing.

In graph form, it looks like this:
<img width="275" alt="image" src="https://user-images.githubusercontent.com/74956050/220544845-d726d1dd-0136-4465-a9f6-d2eaab844105.png">

Does anyone have an idea of why this happens?

**Comment by pusapatiakhilraju** [comment, marked as answer]
The architecture of neural networks in "Workflow.ipynb" does not allow them to learn complex patterns in stock data. In the workflow notebook the neural network architecture consists only of linear layers and do not have activation functions in between the layers to account for non linearity. Hence, increasing the epochs is only one of the solution to improve the performance of the model . There are also many other ways to teach the model different patterns and in the future workbook Daniel specifies what other things could be changed in the architecture to make the model better. Hope this clarifies your question.

**Reply by mrdbourke** [reply]
This is correct!

One neural network doesn't necessarily fit *all* kinds of data.

Especially with stock data (which I wouldn't actually recommend using almost any kind of model to predict, as stocks are mostly unpredictable).

**Reply by ethan530** [reply]
Thanks for replying, the Google data when I used matplotlib to visualise the data looked like this:

<img width="413" alt="image" src="https://user-images.githubusercontent.com/74956050/221115084-5863b073-be22-4e1a-90e4-c1e0ff5c7027.png">

Maybe its just because there are more than 4000 values on the graph, but the training and test data looks surprisingly linear.
